sklearn/neighbors/tests/test_kd_tree.py

```diff
@@ -212,6 +159,7 @@ def test_neighbors_heap(n_pts=5, n_nbrs=10):
 
 
 def test_node_heap(n_nodes=50):
+    rng = np.random.RandomState(42)
```

Comment:
A global mutable RNG was previously reused which is not great with respect to test determinism (given that order of tests can change).